cloud/interfaces.go

```diff
@@ -40,3 +49,18 @@ type CredentialGetter interface {
 	Service
 	GetCredentials(ctx context.Context, group string, cluster string) ([]byte, error)
 }
+
+// Authorizer is an interface which can get the subscription ID and authorizer for an Azure service.
+type Authorizer interface {
+	SubscriptionID() string
```

Comment:
@nader-ziada FYI this is going to conflict with #649 but I can rebase on top of your PR once it lands.

Reply:
no worries, I can also rebase if this gets in first

Reply:
which one will end up in the final result 🙂 I would prefer to keep as struct instead of interface.

Reply:
@alexeldeib can you expand on that a bit?